src/daemon/rpmostree-sysroot-upgrader.c

```diff
@@ -374,58 +374,108 @@ rpmostree_sysroot_upgrader_pull_base (RpmOstreeSysrootUpgrader  *self,
     {
-      if (!ostree_repo_resolve_rev (self->repo, refspec, FALSE, &new_base_rev, error))
-        return FALSE;
+    case RPMOSTREE_REFSPEC_TYPE_OSTREE:
+      {
+        g_autofree char *origin_remote = NULL;
+        g_autofree char *origin_ref = NULL;
+        if (!ostree_parse_refspec (refspec, &origin_remote, &origin_ref, error))
+          return FALSE;
+
+        g_assert (self->origin_merge_deployment);
+        if (origin_remote && !synthetic)
+          {
+            g_autoptr(GVariantBuilder) optbuilder = g_variant_builder_new (G_VARIANT_TYPE ("a{sv}"));
+            if (dir_to_pull && *dir_to_pull)
+              g_variant_builder_add (optbuilder, "{s@v}", "subdir",
+                                     g_variant_new_variant (g_variant_new_string (dir_to_pull)));
+            g_variant_builder_add (optbuilder, "{s@v}", "flags",
+                                   g_variant_new_variant (g_variant_new_int32 (flags)));
+            /* Add the timestamp check, unless disabled. The option was added in
+             * libostree v2017.11 */
+            if (!allow_older)
+              g_variant_builder_add (optbuilder, "{s@v}", "timestamp-check",
+                                     g_variant_new_variant (g_variant_new_boolean (TRUE)));
+            g_variant_builder_add (optbuilder, "{s@v}", "refs",
+                                   g_variant_new_variant (g_variant_new_strv (
+                                                                              (const char *const *)&origin_ref, 1)));
+            if (override_commit)
+              g_variant_builder_add (optbuilder, "{s@v}", "override-commit-ids",
+                                     g_variant_new_variant (g_variant_new_strv (
+                                                                                (const char *const *)&override_commit, 1)));
+
+            g_autoptr(GVariant) opts = g_variant_ref_sink (g_variant_builder_end (optbuilder));
+            if (!ostree_repo_pull_with_options (self->repo, origin_remote, opts, progress,
+                                                cancellable, error))
+              return FALSE;
+
+            if (progress)
+              ostree_async_progress_finish (progress);
+          }
+
+        if (override_commit)
+          new_base_rev = g_strdup (override_commit);
+        else
+          {
+            if (!ostree_repo_resolve_rev (self->repo, refspec, FALSE, &new_base_rev, error))
+              return FALSE;
+          }
+      }
+      break;
+    case RPMOSTREE_REFSPEC_TYPE_ROJIG:
+      {
+        // Not implemented yet, though we could do a query for the provides
+        g_assert (!override_commit);
+
+        g_autoptr(GKeyFile) tsk = g_key_file_new ();
+        g_key_file_set_string (tsk, "tree", "jigdo", refspec);
+        const char *jigdo_version = rpmostree_origin_get_jigdo_version (self->origin);
+        if (jigdo_version)
+          g_key_file_set_string (tsk, "tree", "jigdo-version", jigdo_version);
```

Comment:
The `jigdo-version` stuff isn't hooked up yet on the `DeployTransaction` side, right? E.g. one can't yet do `rpm-ostree rebase updates:$jigdorpm $version`, right?

Reply:
Right, not yet.